scripts/linz-bde-schema-load.bash

```diff
@@ -138,38 +162,44 @@ fi
 
 
 
-if test $DBPATCH_SUPPORTS_STDOUT = yes; then
+if test $DBPATCH_SUPPORTS_STDOUT = yes
+then
     ${DBPATCH_LOADER} ${DBPATCH_OPTS} - _patches || rollback
 fi
 
 # Enable table_version if needed
-if test "${ADD_REVISIONS}" = "yes" -a $TABLEVERSION_SUPPORTS_STDOUT = yes; then
+if test "${ADD_REVISIONS}" = "yes" -a $TABLEVERSION_SUPPORTS_STDOUT = yes
+then
     ${TABLEVERSION_LOADER} ${TABLEVERSION_OPTS} - || rollback
 fi
 
-if test $PGDATABASE != "-"; then
+if test "$PGDATABASE" != "-"
+then
     echo 'SET client_min_messages TO WARNING;'
 fi
 
 cat << EOF
 CREATE EXTENSION IF NOT EXISTS postgis SCHEMA public;
 EOF
 
-for file in ${SCRIPTSDIR}/*.sql; do
+for file in "${SCRIPTSDIR}"/*.sql
+do
     if test ${SKIP_INDEXES} = 'yes' &&
-       `basename $file .sql` = '04-bde_schema_index';
+       [ "$(basename "$file" .sql)" = '04-bde_schema_index' ]
```

Comment:
Ouch, this looks like an actual bug in the original code. It should probably be a separate PR.

Reply:
How so? This change is required for Shellcheck to pass the https://www.shellcheck.net/wiki/SC2283  check

Reply:
The original code was

```shell
     if test ${SKIP_INDEXES} = 'yes' &&
       `basename $file .sql` = '04-bde_schema_index'; 
```

To see the problem, we can simplify that to `test yes = 'yes' && 04-bde_schema_index = '04-bde_schema_index'`. But that fails, because the `&&` is a command separator, and doesn't work the same as `-a` for the `test` command:

```console
$ test yes = 'yes' && 04-bde_schema_index = '04-bde_schema_index'
04-bde_schema_index: command not found
```

So the original code was broken, in the sense that the second line would never be true.